gandi/resource_dnsseckey.go

```diff
@@ -145,9 +134,7 @@ func resourceDNSSECKeyDelete(d *schema.ResourceData, meta interface{}) (err erro
 		parts := strings.SplitN(id, "/", 2)
 		domain = parts[0]
 		id = parts[1]
-		if err = d.Set("id", id); err != nil {
-			return fmt.Errorf("failed to set id for %s: %w", d.Id(), err)
-		}
+		d.SetId(id)
```

Comment:
Same as the above, shouldn't we do error checking?

Reply:
The `SetId` method does not return an error: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema#ResourceData.SetId

Reply:
er oops -- not sure why the linter whined at me lol I guess I should go and clean that up and PR it

Reply:
Actually: 

```
// If the key is invalid or the value is not a correct type, an error
// will be returned.
```

Reply:
It does indirectly actually return an error.

Reply:
Ah okay :)